llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -1290,6 +1312,11 @@ ScheduleDAGInstrs *GCNPassConfig::createMachineScheduler(
   if (EnableMaxIlpSchedStrategy)
     return createGCNMaxILPMachineScheduler(C);
 
+  if (EnableMaxMemoryClauseSchedStrategy ||
+      C->MF->getFunction().hasFnAttribute(
+          "amdgpu-enable-max-memory-clause-scheduling-strategy"))
```

Comment:
probably should avoid the word "enable", and this isn't very flexible in terms of selecting from the chain options we have here. How about an attribute with a named strategy?

Reply:
The suggestion sounds pretty good. I have changed the attribute to named strategy.